dashboard/app/models/script.rb

```diff
@@ -1296,7 +1296,7 @@ def update_text(unit_params, unit_text, metadata_i18n, general_params)
           login_required: general_params[:login_required].nil? ? false : general_params[:login_required], # default false
           wrapup_video: general_params[:wrapup_video],
           family_name: general_params[:family_name].presence ? general_params[:family_name] : nil, # default nil
-          published_state: general_params[:published_state].nil? ? SharedConstants::PUBLISHED_STATE.in_development : general_params[:published_state],
+          published_state: general_params[:published_state],
```

Comment:
the change on this line seems safe because if published state is nil in the database, it will default to in_development when we try to read it: https://github.com/code-dot-org/code-dot-org/blob/20a2975d666fb6dd65ae0475b2b2c8c58603abba/dashboard/app/models/script.rb#L1789-L1792